### PR TITLE
chore: enforce types imports

### PR DESCRIPTION
**Pull request by Yovach**
This PR, however small, adds the keyword 'type' to imports where possible.

I think it would be interesting to enable `verbatimModuleSyntax` as well.

### CODE REVIEW

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks! I think we can use this ESLint rule to enforce it properly: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/consistent-type-imports.md

I personally prefer the `separate-type-imports` `fixStyle`.

**Comment by Yovach** [comment]
I added the `separate-type-imports` rule to `eslint` config and tried to add a rule to enforce the top-level `type` but you have to install a new dependency (https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md#prefer-top-level) :/

File: .eslintrc [hunk 1]
```diff
@@ -14,7 +14,8 @@
   ],
   "rules": {
     "@typescript-eslint/no-var-requires": "off",
-    "@typescript-eslint/no-explicit-any": "off"
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/consistent-type-imports": ["warn", { "fixStyle": "separate-type-imports" }]
```
Comment (QuiiBz) [inline .eslintrc:18]: It should be an error since we want to enforce it.

**Review by QuiiBz — CHANGES_REQUESTED** [review]

**Comment by QuiiBz** [comment]
> tried to add a rule to enforce the top-level type but you have to install a new dependency ([import-js/eslint-plugin-import@main/docs/rules/consistent-type-specifier-style.md#prefer-top-level](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md?rgh-link-date=2023-10-11T11%3A56%3A37Z#prefer-top-level)) :/

This rule seems redundant with `@typescript-eslint/consistent-type-imports`, I'm not sure to understand the difference between the two?

**Comment by Yovach** [comment]
> > tried to add a rule to enforce the top-level type but you have to install a new dependency ([import-js/eslint-plugin-import@main/docs/rules/consistent-type-specifier-style.md#prefer-top-level](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md?rgh-link-date=2023-10-11T11%3A56%3A37Z#prefer-top-level)) :/
> 
> This rule seems redundant with `@typescript-eslint/consistent-type-imports`, I'm not sure to understand the difference between the two?

The `@typescript-eslint/consistent-type-imports` allows us to use
```typescript
import { type ReactElement } from 'react';
```
while it shouldn't.

The `import/consistent-type-specifier-style` prevents this

**Comment by QuiiBz** [comment]
Thanks for the explanation, it's a lot clearer. I think we can go ahed and add this new `eslint-plugin-import` package along with this other rule.

**Comment by Yovach** [comment]
Issues will be fixed with `--fix` thanks to `import/consistent-type-specifier-style`

**Review by QuiiBz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**